server/src/main/java/org/opensearch/common/settings/Setting.java

```diff
@@ -1256,13 +1256,24 @@ private static boolean isFiltered(Property[] properties) {
     public static class RegexValidator implements Writeable, Validator<String> {
         private Pattern pattern;
 
+        private boolean isMatching = true;
+
         /**
          * @param regex A regular expression containing the only valid input for this setting.
          */
         public RegexValidator(String regex) {
             this.pattern = Pattern.compile(regex);
```

Comment:
(Style) This (still) works, but the pattern in most of this class is to overload constructors to call other constructors.  So you could use `this(regex, true)` here, and then not pre-initialize the boolean.

Also probably want to make a similar summary statement as the one you are adding in the constructor below.